Windows Build Broken

Recently the windows build broke. Seems like something must have changed on azures CI imagine that we use because the builds are all ending with a log that says it cant find the compiler:

```
2021-06-02T12:48:17.1298735Z ##[section]Starting: Run CMake to configure build
2021-06-02T12:48:17.1474696Z ==============================================================================
2021-06-02T12:48:17.1475013Z Task         : Command line
2021-06-02T12:48:17.1475309Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2021-06-02T12:48:17.1475594Z Version      : 2.182.0
2021-06-02T12:48:17.1475807Z Author       : Microsoft Corporation
2021-06-02T12:48:17.1476100Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2021-06-02T12:48:17.1476444Z ==============================================================================
2021-06-02T12:48:18.5040686Z Generating script.
2021-06-02T12:48:18.5619267Z ========================== Starting Command Output ===========================
2021-06-02T12:48:18.6082948Z ##[command]"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\c0be6fdb-a8cc-4b97-844b-1dd879010b09.cmd""
2021-06-02T12:48:19.4187605Z **********************************************************************
2021-06-02T12:48:19.4188833Z ** Visual Studio 2019 Developer Command Prompt v16.10.0
2021-06-02T12:48:19.4189619Z ** Copyright (c) 2021 Microsoft Corporation
2021-06-02T12:48:19.4190249Z **********************************************************************
2021-06-02T12:48:21.2449393Z [vcvarsall.bat] Environment initialized for: 'x64'
2021-06-02T12:48:21.6443293Z cmake version 3.20.3
2021-06-02T12:48:21.6443812Z 
2021-06-02T12:48:21.6444319Z CMake suite maintained and supported by Kitware (kitware.com/cmake).
2021-06-02T12:48:22.4292911Z -- The CXX compiler identification is unknown
2021-06-02T12:48:24.4864602Z -- The C compiler identification is unknown
2021-06-02T12:48:24.5178648Z CMake Error at CMakeLists.txt:11 (project):
2021-06-02T12:48:24.5179257Z   The CMAKE_CXX_COMPILER:
2021-06-02T12:48:24.5179883Z 
2021-06-02T12:48:24.5180899Z     C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe
2021-06-02T12:48:24.5181292Z 
2021-06-02T12:48:24.5182898Z   is not a full path to an existing compiler tool.
2021-06-02T12:48:24.5183252Z 
2021-06-02T12:48:24.5183648Z   Tell CMake where to find the compiler by setting either the environment
2021-06-02T12:48:24.5184186Z   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
2021-06-02T12:48:24.5184815Z   to the compiler, or to the compiler name if it is in the PATH.
2021-06-02T12:48:24.5185087Z 
2021-06-02T12:48:24.5185178Z 
2021-06-02T12:48:24.5320395Z CMake Error at CMakeLists.txt:11 (project):
2021-06-02T12:48:24.5321083Z   The CMAKE_C_COMPILER:
2021-06-02T12:48:24.5321218Z 
2021-06-02T12:48:24.5321625Z     C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe
2021-06-02T12:48:24.5321903Z 
2021-06-02T12:48:24.5322345Z   is not a full path to an existing compiler tool.
2021-06-02T12:48:24.5322497Z 
2021-06-02T12:48:24.5322848Z   Tell CMake where to find the compiler by setting either the environment
2021-06-02T12:48:24.5323248Z   variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
2021-06-02T12:48:24.5323666Z   the compiler, or to the compiler name if it is in the PATH.
2021-06-02T12:48:24.5323828Z 
2021-06-02T12:48:24.5323949Z 
2021-06-02T12:48:24.5422712Z -- Configuring incomplete, errors occurred!
2021-06-02T12:48:24.5423520Z See also "D:/a/build/CMakeFiles/CMakeOutput.log".
2021-06-02T12:48:24.5424277Z See also "D:/a/build/CMakeFiles/CMakeError.log".
2021-06-02T12:48:24.7150331Z ##[error]Cmd.exe exited with code '1'.
2021-06-02T12:48:24.7614863Z ##[section]Finishing: Run CMake to configure build
```

@nilsnolde @mloskot any ideas of what to try here?

--- a/.azure-pipelines.yml
+++ b/.azure-pipelines.yml
@@ -56,11 +56,11 @@ jobs:
       - task: Cache@2
         displayName: "Cache build"
         inputs:
-          key: '"msvc-v16.9.0" | build | "$(Build.SourceBranch)"'
+          key: '"msvc-v16.10.0" | build | "$(Build.SourceBranch)"'
           path: "$(BUILD_DIR)"
           restoreKeys: |
-            "msvc-v16.9.0" | build | "$(Build.SourceBranch)"
-            "msvc-v16.9.0" | build
+            "msvc-v16.10.0" | build | "$(Build.SourceBranch)"
+            "msvc-v16.10.0" | build
           cacheHitVar: BUILD_CACHE_RESTORED
 
       - task: UsePythonVersion@0

